packages/framework/fluid-static/src/fluidContainer.ts

```diff
@@ -324,6 +346,10 @@ class FluidContainer<TContainerSchema extends ContainerSchema = ContainerSchema>
 		return this.rootDataObject.initialObjects as InitialObjects<TContainerSchema>;
 	}
 
+	public isReadOnly(): boolean {
+		return this.container.readOnlyInfo.readonly ?? true;
```

Comment:
It looks like undefined is more commonly equated with `false` in container.ts - probably check with @vladsud what he'd recommend doing in the undefined case.

Also, you should make sure that the readonlyChanged event is only emitted when the computed value of this property actually changes (so, for example it shouldn't emit for container.readOnlyInfo.readonly going from `undefined` -> `true` as currently written here since isReadOnly() would return `true` both before and after that state change).  I see some guards in connectionManager.ts that suppress raising the event if the state didn't change, but it's not immediately clear if that will prevent the above possibility.

Reply:
That's part of the reason I prefer keep undefined. This PR is to exposing the readonly information to IFluidContainer rather than fixing any existing logic. And internally we emit the event when trying to set the readonly flag. undefined -> true is a readonly state change (from unknown to true/false).

Reply:
We should be able to resolve this here though, one direction or the other.  The public API surface has a higher quality bar than the legacy API surface given its longer support/compat contract, and also because it needs to be consumable by third parties who we may not be able to directly tutorialize on the quirks of the underlying system.